components/masking_functions/src/masking_functions/registration_routines.cpp

```diff
@@ -0,0 +1,1156 @@
+
+ private:
+  virtual std::size_t min_length() const override { return 14; }
+  virtual std::size_t max_length() const override { return 19; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return x_ascii_masking_char;
+  }
+
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    return masking_functions::mask_inner_alphanum(cs_str, 0, 4, masking_char);
+  }
+};
+
+//
+// mask_pan_relaxed(string, [char])
+//
+class mask_pan_relaxed_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 14; }
+  virtual std::size_t max_length() const override { return 19; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return x_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    return masking_functions::mask_inner_alphanum(cs_str, 6, 4, masking_char);
+  }
+};
+
+//
+// mask_ssn(string, [char])
+//
+class mask_ssn_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 9; }
+  virtual std::size_t max_length() const override { return 11; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return star_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    if (cs_str.get_size_in_characters() == max_length()) {
+      auto sresult = masking_functions::mask_inner(cs_str, 4, 5, masking_char);
+      return masking_functions::mask_inner(sresult, 0, 8, masking_char);
+    } else {
+      return masking_functions::mask_inner_alphanum(cs_str, 0, 4, masking_char);
+    }
+  }
+};
+
+//
+// mask_uk_nin(string, [char])
+//
+class mask_uk_nin_impl final : private mask_impl_base {
+ public:
+  using mask_impl_base::calculate;
+  using mask_impl_base::mask_impl_base;
+
+ private:
+  virtual std::size_t min_length() const override { return 9; }
+  virtual std::size_t max_length() const override { return 11; }
+  virtual std::string_view default_ascii_masking_char() const override {
+    return star_ascii_masking_char;
+  }
+  virtual masking_functions::charset_string process(
+      const masking_functions::charset_string &cs_str,
+      const masking_functions::charset_string &masking_char) const override {
+    return masking_functions::mask_inner_alphanum(cs_str, 2, 0, masking_char);
+  }
+};
+
+//
+// mask_uuid(string, [char])
```

Comment:
Ditto. A description of what this function does (perhaps example) would be nice?

Reply:
Added a detailed description.

Reply:
Thank you!